Automattic/wp-calypso
# Inline Help: Stop browsing to broken route on enter keypress

Currently, if I press <kbd>enter</kbd> when the search field has focus, I end up at an invalid route.

## To Reproduce

* Open inline help
* Hit enter while the input field has focue (with or without entering text first)
* Get redirected to https://wordpress.com/settings/writing/false

## In this change:
* Refactor redux for simpler href access via `getInlineHelpCurrentlySelectedLink` selector
* Move `openResult` to the top level component
* Pass `openResult` to the search card & results components
* Pass the href from the child components into the provided `openResult` fn
* Use `lodash#get` throughout the selector to guard against fatals
* Prettify along the way

## To Test

* Run this branch
* Open inline help
* Click in the search box
* Hit the enter key (with and without text present in the field)
  * It should not redirect to an invalid route & searching should work as before
  * No `calypso_inlinehelp_link_open` events should be triggered
* Press the up and down arrow keys to "select" a result
* Hit the enter key
  * It should redirect to the selected support article
  * A properly-formed `calypso_inlinehelp_link_open` events should be triggered
* Go back to calypso and open inline help
* Clicking a result (with and without having searched or selected an item) should trigger a properly-formed `calypso_inlinehelp_link_open` event & redirect to the support article
